cocotb/share/include/embed.h

```diff
@@ -30,6 +30,20 @@
 #ifndef COCOTB_EMBED_H_
 #define COCOTB_EMBED_H_
 
+#ifdef COCOTB_EXPORTS
+#if defined(__linux__) || defined(__APPLE__)
+#define COCOTB_EXPORT
+#else
+#define COCOTB_EXPORT __declspec(dllexport)
+#endif
+#else
+#if defined(__linux__) || defined(__APPLE__)
+#define COCOTB_EXPORT
+#else
+#define COCOTB_EXPORT __declspec(dllimport)
+#endif
+#endif
+
```

Comment:
Can we define this in a single header and use it across all our libraries?

Reply:
If I am remember correctly, these macros once existed in cocotb_util.h, but were removed because they weren't used.

Reply:
They distinct from each other. They are needed for every library that is build. There are only as many as we have libraries. For an explanation of their function, see my comment [above](https://github.com/cocotb/cocotb/pull/1796#issuecomment-627261359).